evutil.c

```diff
@@ -406,13 +406,46 @@ evutil_win_socketpair(int family, int type, int protocol,
 int
 evutil_socketpair(int family, int type, int protocol, evutil_socket_t fd[2])
 {
+	int ret = 0;
 #if defined(_WIN32)
-	return evutil_win_socketpair(family, type, protocol, fd);
+	type &= ~(EVUTIL_SOCK_NONBLOCK|EVUTIL_SOCK_CLOEXEC);
+	ret = evutil_win_socketpair(family, type, protocol, fd);
 #elif defined(EVENT__HAVE_SOCKETPAIR)
-	return socketpair(family, type, protocol, fd);
+	ret = socketpair(family, type, protocol, fd);
 #else
-	return evutil_ersatz_socketpair_(family, type, protocol, fd);
+	ret = evutil_ersatz_socketpair_(family, type, protocol, fd);
+#endif
+	if (ret)
+		return ret;
+#ifndef SOCK_NONBLOCK
+	if (type & EVUTIL_SOCK_NONBLOCK) {
```

Comment:
Hm, actually it is seems that it is always defined - https://github.com/libevent/libevent/blob/f2b3ce6b5587efb206bc74298d7a835372ce2caf/util-internal.h#L531-L540 (hope that there is no `#ifndef _WIN32` that I skipped)

Reply:
So will this work correctly on windows?

Reply:
You're right, I'll fix it.

Reply:
Yes, they are always defined, so this means that the code below does not make sense

As for the windows, I guess it is OK to fail, BTW if you will update all users to `evutil_socketpair(SOCK_NONBLOCK)` then I guess something will stop working if on windows this will fail.

Also, does the value on windows and unix platform matches?

Reply:
There are `SOCK_NONBLOCK` and `SOCK_CLOEXEC` on Windows (if my memory serves me right), and I also didn't try to use them on Windows, I made it fall back to the old approach before this PR by leading the code compiling on Windows to go into `#ifndef SOCK_NONBLOCK` and `#ifndef SOCK_CLOEXEC`.

I have a feeling that we're not on the same page here :)

Reply:
Besides, everything should be still working as it did before: new codes that pass `EVUTIL_SOCK_NONBLOCK`/`EVUTIL_SOCK_CLOEXE` to `evutil_socketpair()` will conserve extra system calls while old codes that are not aware of `EVUTIL_SOCK_NONBLOCK`/`EVUTIL_SOCK_CLOEXE` will continue to work as expected.

Reply:
Oh, now I see, you are using `SOCK_NONBLOCK` for check, so it is OK. Let's also put a comment why SOCK_NONBLOCK is checked instead of EVUTIL_SOCK_NONBLOCK (since the later is always defined)

Reply:
Done.